File: src/Stronghold.Application/Features/GymVisits/GetEligibleForCheckIn/EligibleMemberResponse.cs
namespace Stronghold.Application.Features.GymVisits.GetEligibleForCheckIn;

public class EligibleMemberResponse
{
    public int UserId { get; set; }
    public string UserFullName { get; set; } = string.Empty;
    public string MembershipPackageName { get; set; } = string.Empty;
}

File: stronghold_desktop/lib/features/gym/models/eligible_member_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'eligible_member_response.freezed.dart';
part 'eligible_member_response.g.dart';

@freezed
abstract class EligibleMemberResponse with _$EligibleMemberResponse {
  const factory EligibleMemberResponse({
    required int userId,
    required String userFullName,
    required String membershipPackageName,
  }) = _EligibleMemberResponse;

  factory EligibleMemberResponse.fromJson(Map<String, dynamic> json) =>
      _$EligibleMemberResponseFromJson(json);
}
